src/Security/RolesBuilder/SecurityRolesBuilder.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Security\RolesBuilder;
+
+use Sonata\AdminBundle\Admin\Pool;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * @author Silas Joisten <silasjoisten@hotmail.de>
+ */
+final class SecurityRolesBuilder implements RolesBuilderInterface
+{
+    /**
+     * @var AuthorizationCheckerInterface
+     */
+    private $authorizationChecker;
+
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var string []
+     */
+    private $rolesHierarchy;
+
+    public function __construct(
+        AuthorizationCheckerInterface $authorizationChecker,
+        Pool $pool,
+        TranslatorInterface $translator,
+        array $rolesHierarchy = []
+    ) {
+        $this->authorizationChecker = $authorizationChecker;
+        $this->pool = $pool;
+        $this->translator = $translator;
+        $this->rolesHierarchy = $rolesHierarchy;
+    }
+
+    public function getPermissionLabels(): array
+    {
+        return [];
+    }
```

Comment:
Don't like this empty return, can we somehow segregate interface and avoid it, cc @greg0ire

Reply:
Try if you have a better soulution or tell me than i can try to do this :)

Reply:
I'm on it, let's try, I'm not sure